contrib/platform/src/com/sun/jna/platform/win32/Winevt.java

```diff
@@ -0,0 +1,1705 @@
+                    } else {
+                        return new WinNT.HANDLE(field1.getPointer().getPointer(0));
+                    }
+                default:
+                    throw new IllegalStateException(String.format("NOT IMPLEMENTED: getValue(%s) (Array: %b, Count: %d)", type, isArray(), Count));
+            }
+        }
+    }
+
+    /**
+     * Defines the types of authentication that you can use to authenticate the user when connecting to a remote
+     * computer.
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/aa385570(v=vs.85).aspx
+     */
+    public static interface EVT_RPC_LOGIN_FLAGS {
+
+        /**
+         * Use the default authentication method during RPC login. The default authentication method is Negotiate.
+         */
+        public static final int EvtRpcLoginAuthDefault = 0;
+
+        /**
+         * Use the Negotiate authentication method during RPC login. The client and server negotiate whether to use
+         * NTLM or Kerberos.
+         */
+        public static final int EvtRpcLoginAuthNegotiate = 1;
+
+        /** Use Kerberos authentication during RPC login. */
+        public static final int EvtRpcLoginAuthKerberos = 2;
+
+        /** Use NTLM authentication during RPC login. */
+        public static final int EvtRpcLoginAuthNTLM = 3;
+    }
+
+    /**
+     * Contains the information used to connect to a remote computer.
+     * https://msdn.microsoft.com/en-us/library/windows/desktop/aa385566(v=vs.85).aspx
+     */
+    public class EVT_RPC_LOGIN extends Structure {
+
+        /** The name of the remote computer to connect to. */
+        public String Server;
+
+        /** The user name to use to connect to the remote computer. */
+        public String User;
+
+        /** The domain to which the user account belongs. Optional. */
+        public String Domain;
+
+        /** The password for the user account. */
+        public String Password;
+
+        /**
+         * The authentication method to use to authenticate the user when connecting to the remote computer.
+         * For possible authentication methods, see the {@link Winevt.EVT_RPC_LOGIN_FLAGS} enumeration.
+         */
+        public int Flags;
+
+        public EVT_RPC_LOGIN() {
+            super(W32APITypeMapper.UNICODE);
+        }
+
+        protected List<String> getFieldOrder() {
+            return Arrays.asList("Server", "User", "Domain", "Password", "Flags");
+        }
+
+        public EVT_RPC_LOGIN(String Server, String User, String Domain, String Password, int Flags) {
+            super(W32APITypeMapper.UNICODE);
+            this.Server = Server;
+            this.User = User;
+            this.Domain = Domain;
+            this.Password = Password;
+            this.Flags = Flags;
+        }
+
+        public EVT_RPC_LOGIN(Pointer peer) {
+            super(peer, Structure.ALIGN_DEFAULT, W32APITypeMapper.UNICODE);
+        }
+
+        protected ByReference newByReference() {
```

Comment:
Please check if the the methods newByReference, newByValue and newInstance are needed. I see no reason for their implementation.